libllvm-pdb-wrapper/wrapper.cpp

```diff
@@ -340,6 +344,18 @@ EXPORT void *PDB_File_Create(int Is64Bit) {
     }
     return pdb;
 }
+EXPORT void *PDB_File_Create2(int Is64Bit,uint32_t Age,uint32_t Signature,uint8_t * GUIDData) {
```

Comment:
`uint8_t * GUIDData` should be followed by size_t GUIDSize. Just passing raw pointers is bad practice if especially since it's not really documented *what* to pass exactly here.

Reply:
Wouldn't it be more sensible to change the signature of `PDB_File_Create` to contain the new parameters and only set them if they're != 0? It's been a while but I remember those parameters being optional. Did LLVM change since then?

Reply:
> 
> 
> `uint8_t * GUIDData` should be followed by size_t GUIDSize. Just passing raw pointers is bad practice if especially since it's not really documented _what_ to pass exactly here.
the value that should be passed is this 
```
/// This represents the 'GUID' type from windows.h.
struct GUID {
  uint8_t Guid[16];
};
``` 
copied from `llvm/DebugInfo/CodeView/GUID.h`

a crate that covers the guid type is https://docs.rs/uuid/0.8.2/uuid/

Reply:
> 
> 
> Wouldn't it be more sensible to change the signature of `PDB_File_Create` to contain the new parameters and only set them if they're != 0? It's been a while but I remember those parameters being optional. Did LLVM change since then?

Well I created the new function as to avoid breaking any other projects which may use your code.

Reply:
> Well I created the new function as to avoid breaking any other projects which may use your code.

I don't think it's reasonable to expect this project to support all sorts of LLVM versions as the API fluctuates a lot and isn't really documented. I'm fairly certain that it doesn't even support anything below the currently used version. I think the better way to move forward is to create a version of the package on crates.io of the current version and then to push your changes into a new version.